ledger/sandbox/src/main/scala/com/digitalasset/platform/index/ReadOnlySqlLedger.scala

```diff
@@ -67,7 +67,12 @@ object ReadOnlySqlLedger {
             Future.failed(
               new LedgerIdMismatchException(foundLedgerId, initialLedgerId) with StartupException)
           case None =>
-            Future.successful(initialLedgerId)
+            logger.info("Ledger ID not found in the index database. Retrying again in 5 seconds.")
+            val promise = Promise[LedgerId]()
+            mat.scheduleOnce(5.seconds, () => {
+              promise.completeWith(verifyLedgerId(ledgerDao, initialLedgerId))
+            })
+            promise.future
```

Comment:
Does it make sense to handle the retry at this level? Would it make more sense to throw an exception here and let the caller decide what to do with this?

Reply:
So the question is how we control the retry mechanism. At this point, we know fairly well what went "wrong".

Alternatively we could slap a retry around the entire index initialization process and not really care about specific errors.

What do you think?

Reply:
I'm only a bit concerned of ending up with retry logic scattered across many classes. We can still signal a specific failure using a specific exception and handle it appropriately. Feel free to leave this here if you think it's better to deal with failures as close as possible to the source.

Reply:
If we do decide to leave the retry here, should we stop retrying at some point?

Reply:
@stefanobaghino-da: I'm open to suggestions on where initialization errors should be handled. I think this falls into a similar category as waiting for the DB to be available (see #6146). There we currently retry 600 times with a retry delay of 1 second. We can do something similar here.

Since this is only run on startup I think it makes sense to have some leniency during the startup sequence between a read-only api server and the indexer, so not failing the entire process.